pkg/tcpip/network/ipv4/ipv4.go

```diff
@@ -105,6 +107,9 @@ func (e *endpoint) PrefixLen() int {
 // MaxHeaderLength returns the maximum length needed by ipv4 headers (and
 // underlying protocols).
 func (e *endpoint) MaxHeaderLength() uint16 {
+	if len(options) > 0 {
```

Comment:
options?

Reply:
IPv4 HeaderLength will vary if options present so before returning MaxHeaderLength, we are checking options as to be included or not and returning correct length.